.github/actions/check-changes/action.yaml

```diff
@@ -0,0 +1,57 @@
+name: Check For Changed Files
+description: Checks for changed files compared to specific git reference and glob expression
+inputs:
+  baseRef:
+    description: Git reference to check against
+    required: true
+  ref:
+    description: Git reference to check with
+    required: false
+    default: HEAD
+  checkGlob:
+    description: Glob expression to limit check to specific files
+    required: false
+  useFallback:
+    description: Use fallback compare against prior commit
+    required: false
+    default: 'true'
+outputs:
+  passed:
+    value: ${{ steps.checks.outputs.passed }}
+    description: True if specified files where changed in comparison to specified git reference
```

Comment:
The job checks if changes to the specified files exist. As such the check "passes" if changes are found. The explanation is handled by the `description`.

Reply:
Use of the word "passed" Is strange because `passed=true` if files have changed. But if files have changed, we actually consider this a failed state, no?

Reply:
The check is self-contained, so it makes no assumption about what file changes might mean.

That is a concern of the workflow that _calls_ the action, so from the action that checks for changed files, detected changes are a _success_ (because otherwise the check would need to be called "check-clean" or something).

Reply:
To me, the name `passed` is a cognitive disconnect. `filesChanged=true` would be fine because it's obviously communicating the intent of the boolean.
This logic, in general, is fine:
```
if (checkChanges)
// something is wrong and warrants action
else()
// everything is okay and there is nothing to do
```

This specific naming, however, warrants a double-take from me:
```
if (steps.checks.outputs.passed)
// something is wrong and warrants action
else()
// everything is okay and there is nothing to do
```

It's not the boolean true implying failure that bothers me the most here, it's the boolean name itself.

Hope that clarifies what I was trying to say. I'll probably not block on this, even if I do find it confusing.

Reply:
I'm not opposed to changing the term, but given that it checks for changed files, and it's "successful" when changes are detected, what would be the appropriate term for a positive check?

Reply:
I'd have gone with `filesChanged` instead of `passed`. If you want something more easily distinguished from `changedFiles`, then perhaps `areFilesChanged` or `hasChangedFiles`. Feel free to seek another opinion in case this is just something that bothers only me.

Reply:
That would make the two outputs "filesChanged" and "changedFiles" - "hasChangedFiles" would be canonical with the output in the services validator.